### PR TITLE
feat: add privateCa to kaniko pods

### PR DESCRIPTION
**Pull request by ThibaultFy**
## Description

closes FL-1485

## Checklist

- [ ] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: backend/builder/image_builder/image_builder.py [hunk 1]
```diff
@@ -284,6 +295,10 @@ def _build_container(dockerfile_mount_path: str, image_tag: str) -> kubernetes.c
         )
         volume_mounts.append(docker_config)
 
+    if PRIVATE_CA_ENABLED:
+        docker_config = kubernetes.client.V1VolumeMount(name=CA_SECRET_NAME, mount_path="/kaniko/ssl/certs")
```
Comment (ThibaultFy) [inline backend/builder/image_builder/image_builder.py:299]: @guilhem-barthes how the system knows that it must calls the file `ca-certificates.crt` ?

Comment (guilhem-barthes) [reply backend/builder/image_builder/image_builder.py:299]: That's the default for ssh configuration

**Review by thbcmlowk — APPROVED** [review]
LGTM! Great work 🔏

**Pull request merged**